azurerm/resource_arm_servicebus_topic.go

```diff
@@ -193,41 +191,40 @@ func resourceArmServiceBusTopicRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("name", resp.Name)
 	d.Set("resource_group_name", resGroup)
 	d.Set("namespace_name", namespaceName)
-	d.Set("location", azureRMNormalizeLocation(*resp.Location))
-
-	props := resp.TopicProperties
-	d.Set("status", string(props.Status))
-	d.Set("auto_delete_on_idle", props.AutoDeleteOnIdle)
-	d.Set("default_message_ttl", props.DefaultMessageTimeToLive)
-
-	if props.DuplicateDetectionHistoryTimeWindow != nil && *props.DuplicateDetectionHistoryTimeWindow != "" {
-		d.Set("duplicate_detection_history_time_window", props.DuplicateDetectionHistoryTimeWindow)
-	}
-
-	d.Set("enable_batched_operations", props.EnableBatchedOperations)
-	d.Set("enable_express", props.EnableExpress)
-	d.Set("enable_filtering_messages_before_publishing", props.FilteringMessagesBeforePublishing)
-	d.Set("enable_partitioning", props.EnablePartitioning)
-	d.Set("requires_duplicate_detection", props.RequiresDuplicateDetection)
-	d.Set("support_ordering", props.SupportOrdering)
 
-	maxSize := int(*props.MaxSizeInMegabytes)
+	if props := resp.SBTopicProperties; props != nil {
+		d.Set("status", string(props.Status))
+		d.Set("auto_delete_on_idle", props.AutoDeleteOnIdle)
+		d.Set("default_message_ttl", props.DefaultMessageTimeToLive)
 
-	// if the topic is in a premium namespace and partitioning is enabled then the
-	// max size returned by the API will be 16 times greater than the value set
-	if *props.EnablePartitioning {
-		namespace, err := meta.(*ArmClient).serviceBusNamespacesClient.Get(resGroup, namespaceName)
-		if err != nil {
-			return err
+		if props.DuplicateDetectionHistoryTimeWindow != nil && *props.DuplicateDetectionHistoryTimeWindow != "" {
+			d.Set("duplicate_detection_history_time_window", props.DuplicateDetectionHistoryTimeWindow)
 		}
 
-		if namespace.Sku.Name != servicebus.Premium {
-			const partitionCount = 16
-			maxSize = int(*props.MaxSizeInMegabytes / partitionCount)
+		d.Set("enable_batched_operations", props.EnableBatchedOperations)
+		d.Set("enable_express", props.EnableExpress)
+		d.Set("enable_partitioning", props.EnablePartitioning)
+		d.Set("requires_duplicate_detection", props.RequiresDuplicateDetection)
+		d.Set("support_ordering", props.SupportOrdering)
+
+		maxSize := int(*props.MaxSizeInMegabytes)
+
+		// if the topic is in a premium namespace and partitioning is enabled then the
+		// max size returned by the API will be 16 times greater than the value set
+		if *props.EnablePartitioning {
+			namespace, err := meta.(*ArmClient).serviceBusNamespacesClient.Get(resGroup, namespaceName)
+			if err != nil {
+				return err
+			}
+
+			if namespace.Sku.Name != servicebus.Premium {
```

Comment:
I might be reading this wrong but I just want to confirm based on your comment that premium has max size * 16

Reply:
this has been around for a while - so it hasn't changed AFAIK